src/server/services/pipeline-in-progress/validateReadings.js

```diff
@@ -32,14 +36,18 @@ function validateReadings(arrayToValidate, conditionSet, meterIdentifier = undef
  * @param {Moment} maxDate inclusive latest acceptable date (won't be rejected)
  * @param {number} maxError maximum number of errors to be reported, ignore the rest
  * @param {string} meterIdentifier identifier of meter being checked.
+ * @returns {object} { validDates, invalidDates, errMsg }
  */
 function checkDate(arrayToValidate, minDate, maxDate, maxError, meterIdentifier) {
-	let validDates = true;
+	let validDates = [];
+	let invalidDates = [];
 	let errMsg = '';
+	let readingNumber = 0;
+
```

Comment:
This is the last one I'm marking but OED only puts blank lines to separate important block and avoid otherwise.